### PR TITLE
Make ExtraNodesetGenerator work with distributed mesh

### LINKED ISSUE #15728: ExtraNodesetGenerator does not work with distributed mesh
## Bug Description
When specify IDs of nodes to be added in the node set, nodes with these IDs may only be on one processor, thus make call `BoundaryInfo::add_node` error out.

## Steps to Reproduce
All current tests for this mesh generator are restricted with replicated mesh. NS module uses this to pin pressure at a node. If this cannot be fixed, those NS tests cannot run with distributed mesh. It might be an easy fix by protecting `add_node` call by querying if the node exist on the mesh.

## Impact
Enable tests using this mesh generator to run in parallel with distributed mesh.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #15728.

### CODE REVIEW

File: framework/src/meshgenerators/ExtraNodesetGenerator.C [hunk 1]
```diff
@@ -78,7 +78,11 @@ ExtraNodesetGenerator::generate()
   // add nodes with their ids
   for (const auto & node_id : nodes)
```
Comment (loganharbour) [inline framework/src/meshgenerators/ExtraNodesetGenerator.C:79]: Can you do this instead?

```
for (const auto & node_id : nodes)
{
  const Node * node_ptr = mesh->query_node_ptr(node_id);
  if (node_ptr)
    for (const auto boundary_id : boundary_ids)
      boundary_info.add_node(node_ptr, boundary_id);
}
```

We're adding boundary_ids.size() extra node queries with what's currently there.

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — APPROVED** [review]
Thanks @YaqiWang 👍🏼 

I'm good with this when it's green.

**Comment by YaqiWang** [comment]
Why is this still testing?!

**Review by loganharbour — APPROVED** [review]
This passed: https://civet.inl.gov/pr/11592/

The CIVET hook to updating github wasn't working too well yesterday.

**Comment by YaqiWang** [comment]
Thanks @loganharbour.

**Pull request merged**